### PR TITLE
added user watch endpoint

### LINKED ISSUE #5564: Add watch endpoint for User/UserSettings
Add a watch endpoint for User, and with it UserSettings which would be used by the UI to quicky recognize the usersetting changes and apply them.

### PR DESCRIPTION
**Pull request by lsviben**
**What this PR does / why we need it**:
adds a user watch websocket endpoint. Primarily it will be used by the UI to catch the user settings changes and apply them. 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #5564 

**Does this PR introduce a user-facing change?**:

```release-note
added a websocket user self-watch endpoint 
```

### CODE REVIEW

File: pkg/watcher/kubernetes/user.go [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+		provider:  provider,
+		watcher:   watcher,
+		publisher: pubsub.New(),
+	}
+
+	go w.run()
+	return w, nil
+}
+
+// run and publish information about settings updates. Watch will restart itself if any error occurs.
```
Comment (maciaszczykm) [inline pkg/watcher/kubernetes/user.go]: Comment needs update.

**Comment by maciaszczykm** [comment]
LGTM after you update the mentioned comment. Tested locally and it works fine :)

**Comment by maciaszczykm** [comment]
/lgtm

@lsviben Please avoid squashing commits in pull requests as it makes the review harder. The bot will do it anyways :)

**Comment by lsviben** [comment]
> /lgtm
> 
> @lsviben Please avoid squashing commits in pull requests as it makes the review harder. The bot will do it anyways :)

Didnt know that, thanks for the info :)

**Pull request merged**